test/ui-testing/tags-crud.js

```diff
@@ -0,0 +1,131 @@
+          .wait('#tags_enabled')
+          .wait(222)
+          .evaluate(() => {
+            const element = document.querySelector('[value="true"]');
+            element.click();
+          })
+          .then(() => {
+            nightmare
+              .wait(222)
+              .wait('#tags_enabled')
+              .click('#tags_enabled')
+              .wait('#clickable-save-config')
+              .click('#clickable-save-config')
+              .then(done)
+              .catch(done);
+          });
+      });
+
+      it('should open Agreements app', done => {
+        helpers.clickApp(nightmare, done, 'agreements');
+      });
+
+      it('should confirm correct URL', done => {
+        nightmare
+          .evaluate(() => document.location.pathname)
+          .then(pathName => {
+            if (!pathName.includes('/erm/agreements')) throw Error('URL is incorrect');
+            done();
+          })
+          .catch(done);
+      });
+
+      it('should create agreement', done => {
+        AgreementCRUD.createAgreement(nightmare, done, agreement);
+      });
+
+      it(`should search for and find agreement: ${agreement.name}`, done => {
+        nightmare
+          .wait('#input-agreement-search')
+          .insert('#input-agreement-search', agreement.name)
+          .click('[data-test-search-and-sort-submit]')
+          .wait(1000) // If another agreement was open wait for the new one to be open before the next operation.
+          .wait('#agreementInfo')
+          .evaluate(expectedValues => {
+            const node = document.querySelector('[data-test-agreement-name]');
+            if (!node || !node.innerText) throw Error('No agreement name node found.');
+
+            const name = node.innerText;
+            if (name !== expectedValues.name) {
+              throw Error(`Name of found agreement is incorrect. Expected "${expectedValues.name}" and got "${name}" `);
+            }
+          }, agreement)
+          .then(done)
+          .catch(done);
+      });
+
+      it('should add a new Tag', done => {
+        nightmare
+          .wait('#clickable-show-tags')
+          .click('#clickable-show-tags')
+          .wait('#input-tag-input')
+          .type('#input-tag-input', testTag)
+          .click('#input-tag-input')
+          .wait(2222)
+          .type('#input-tag-input', '\u000d')
+          .wait((tagValue) => {
+            return Array.from(
+              document.querySelectorAll('div[class*="valueChipValue"]')
+            ).findIndex(e => e.textContent === tagValue) >= 0;
+          }, testTag)
+          .then(() => {
+            nightmare
+              .then(done)
+              .catch(done);
+          })
+          .catch(done);
+      });
+
+      it('should have the badge count equal to 1', done => {
+        nightmare
+          .wait('#clickable-show-tags')
+          .wait(() => {
+            return Array.from(
+              document.querySelectorAll('span[class*="label"]')
```

Comment:
Way too broad a selector here. Not only will this pass if any badge has a value of 1, it'll pass if any span with a class containing `label` (pretty common) _on the page_ has content of `1`. I think we can make this more specific.

Reply:
Yeah, right. Replacing it with
 _.evaluate(() => parseInt(document.querySelector('#clickable-show-tags').textContent) === 1)_